metals/src/main/scala/scala/meta/internal/bsp/MetalsBspException.scala

```diff
@@ -1,6 +1,10 @@
 package scala.meta.internal.metals
 
-case class MetalsBspException(tryingToGet: String, msg: String)
-    extends Exception(
-      s"BSP connection failed in the attempt to get: $tryingToGet.  $msg"
+class MetalsBspException(
+    tryingToGet: String,
+    msg: String,
+    cause: Throwable = null
```

Comment:
Why the change from case class to class? Also if we are storing both the `msg` and the `cause` couldn't we just store the `cause` and pull the `msg` from it when needed? I think we should also avoid the `null` default parameter unless there is a good reason to use it.

Reply:
> Why the change from case class to class?

Exceptions shouldn't be case classes, because of their attached stack trace. The case class `equals` method doesn't take the stack trace into account, and wrongfully assumes different stack traces to be equal.

If `MetalsBspException` is a case class, we have
```scala
val ex1 = MetalsBspException("a", "b") // line 1
val ex2 = MetalsBspException("a", "b") // line2
ex1.getStackTrace.toSeq == ex2.getStackTrace.toSeq // false, ex1 has line 1, ex2 has line 2
ex1 == ex2 // wrongfully true
```
(`ex1` and `ex2` have different stack traces, but are made equals nonetheless.)

Reply:
> Also if we are storing both the `msg` and the `cause` couldn't we just store the `cause` and pull the `msg` from it when needed?

Good point.

Reply:
> I think we should also avoid the null default parameter unless there is a good reason to use it.

Just removed the default value, it wasn't necessary.

Reply:
> Exceptions shouldn't be case classes, because of their attached stack trace. The case class equals method doesn't take the stack trace into account, and wrongfully assumes different stack traces to be equal.

Ahhh of course. 👍🏼 thanks for the explanation.

Reply:
(Also, I wanted to mention: I changed this so that I can detect an `InterruptedException` in the cause in the added test.)